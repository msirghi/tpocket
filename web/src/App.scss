@import "sass/main";

* {
    margin: 0;
    padding: 0;
}

