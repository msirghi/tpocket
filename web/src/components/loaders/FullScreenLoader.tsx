import React from 'react';
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '90vw',
      height: '90vh',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    },
  })
);

export const FullScreenLoader = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress size={80} />
    </div>
  );
};
