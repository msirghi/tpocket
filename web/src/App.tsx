import React, { useEffect, useState } from 'react';
import { Routes } from './Routes';
import { setAccessToken } from './accessToken';
import { FullScreenLoader } from './components/loaders/FullScreenLoader';

export const App: React.FC<{}> = () => {
  const [loading, setLoading] = useState<boolean>(true);

  useEffect(() => {
    fetch(`http://localhost:4000/refresh_token`, { method: 'POST', credentials: 'include' }).then(
      async (res) => {
        const { accessToken } = await res.json();
        if (!accessToken && !window.location.href.includes('login')) {
          window.location.replace('http://localhost:3000/login');
        }
        setAccessToken(accessToken);
        setLoading(false);
      }
    );
  }, []);

  if (loading) {
    return <FullScreenLoader />;
  }

  return <Routes />;
};
