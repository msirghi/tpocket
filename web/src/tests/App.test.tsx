import { shallow, ShallowWrapper, mount } from 'enzyme';
import { App } from '../App';
import React from 'react';
import { FullScreenLoader } from '../components/loaders/FullScreenLoader';
import toJson from 'enzyme-to-json';

describe('App component', () => {
  let component: ShallowWrapper;

  afterAll(() => {
    jest.clearAllMocks();
  });

  beforeEach(() => {
    component = shallow(<App />);
  });

  it('should make a snaphost of a loader on initial rendering', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should show FullScreenLoader component on loading', () => {
    const loader = component.find(FullScreenLoader);
    expect(loader).toHaveLength(1);
  });

  it('should perform a request on page load', async () => {
    fetch.mockResponseOnce(JSON.stringify({ accessToken: 'accessToken' }));
    mount(<App />);
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toHaveBeenCalledWith('http://localhost:4000/refresh_token', {
      credentials: 'include',
      method: 'POST'
    });
  });

  it('should fire window.location.replace on page render and access token null', () => {
    window.location.href = 'page';
    fetch.mockResponseOnce(JSON.stringify({ accessToken: null }));
    jest.spyOn(window.location, 'replace');

    mount(<App />);
    spyOn(window.location, 'replace');
    expect(fetch).toHaveBeenCalled();
    expect(jest.isMockFunction(window.location.replace)).toBe(true);
  });
});
