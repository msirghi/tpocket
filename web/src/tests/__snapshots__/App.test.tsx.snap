// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should make a snaphost of a loader on initial rendering 1`] = `<FullScreenLoader />`;
